django/db/migrations/autodetector.py

```diff
@@ -225,14 +226,11 @@ def _generate_through_model_map(self):
             old_model_name = self.renamed_models.get((app_label, model_name), model_name)
             old_model_state = self.from_state.models[app_label, old_model_name]
             for field_name, field in old_model_state.fields:
-                old_field = self.old_apps.get_model(app_label, old_model_name)._meta.get_field(field_name)
-                if (hasattr(old_field, "remote_field") and getattr(old_field.remote_field, "through", None) and
-                        not old_field.remote_field.through._meta.auto_created):
-                    through_key = (
-                        old_field.remote_field.through._meta.app_label,
-                        old_field.remote_field.through._meta.model_name,
-                    )
-                    self.through_users[through_key] = (app_label, old_model_name, field_name)
+                if hasattr(field, 'remote_field') and getattr(field.remote_field, 'through', None):
+                    through_key = resolve_model_key(app_label, model_name, field.remote_field.through)
+                    through_model_state = self.from_state.models[through_key]
+                    if not through_model_state.options.get("auto_created"):
+                        self.through_users[through_key] = (app_label, old_model_name, field_name)
```

Comment:
Are you sure this branch is ever skipped? AFAIK `auto_created` models are not part `ProjectState.models` entries.

Reply:
You're most probably right :+1: I'll remove those conditions